src/utils.js

```diff
@@ -64,6 +64,83 @@ CKEDITOR.define( function() {
 		 */
 		isObject: function( obj ) {
 			return typeof obj === 'object' && !!obj;
-		}
+		},
+
+		/**
+		 * A mixin function to be used to implement the static `extend()` method in classes. It allows for easy creation
+		 * of subclasses.
+		 *
+		 * @param {Object} [proto] Extensions to be added to the subclass prototype.
+		 * @param {Object} [statics] Additional static properties to be added to the subclass constructor.
+		 * @returns {Object} When executed as a static method of a class, it returns the new subclass constructor.
+		 */
+		extendMixin: function( proto, statics ) {
+			var that = this;
+			var child = ( proto && proto.hasOwnProperty( 'constructor' ) ) ?
+					proto.constructor :
+					function() {
+						that.apply( this, arguments );
+					};
+
+			// Copy the statics.
+			utils.extend( child, this, statics );
+
+			// Use the same prototype.
+			child.prototype = Object.create( this.prototype );
+
+			// Add the new prototype stuff.
+			if ( proto ) {
+				utils.extend( child.prototype, proto );
+			}
+
+			return child;
+		},
+
+		/**
+		 * Creates a spy function (ala Sinon.js) that can be used to inspect call to it.
+		 *
+		 * The following are the present features:
+		 *
+		 *  * spy.called: property set to `true` if the function has been called at least once.
+		 *
+		 * @returns {Function} The spy function.
+		 */
+		spy: function() {
+			var spy = function() {
+				spy.called = true;
+			};
+
+			return spy;
+		},
+
+		/**
+		 * Returns a unique id. This id is a number (starting from 1) which will never get repeated on successive calls
+		 * to this method.
+		 *
+		 * @returns {Number} A number representing the id.
+		 */
+		uid: ( function() {
+			var next = 1;
+
+			return function() {
+				return next++;
+			};
+		} )()
 	};
+
+	return utils;
 } );
+
+/**
```

Comment:
This is not required. You can inherit docs of any method of every class.

But I would actually do this differently. I think that we need a `Class` mixin and this mixin will expose `extend` static method. Then every class will contain `@mixins Class` and voila. `Class` could also have other methods like `_def()` for easier `Object.defineProperty()`. Plus it could do the boilerplate stuff:

```
Model.extend = utils.extendMixin;

utils.extend( Model.prototype, EmitterMixin, { ... } );
```

Basically - it's time to discuss if we want the `Class` (or however you want to call it) to exist or not. I think that there are reasons for having it, but I'm not sure about the shape of it.